avocado/core/task/statemachine.py

```diff
@@ -199,6 +245,11 @@ async def monitor(self):
                 runtime_task.status = 'FINISHED'
             except asyncio.TimeoutError:
                 runtime_task.status = 'FINISHED W/ TIMEOUT'
+        task = runtime_task.task
+        task_data = self._state_machine._status_repo.get_latest_task_data(str(task.identifier))
```

Comment:
Isn't there a possibility here that we miss this from the **latest** task data?  I have the feeling that this would be more fail proof if we look at the overall stats, like:

```python
if 'fail' in self._state_machine._status_repo.result_stats:
```

And BTW, we should check/revisit if we should only cover `fail` or every "non success" or "neutral" (like cancel/skip) result.

Reply:
I will use result_stats. Thanks.

About, "non success", since the flag is called "failfast" and the documentation says "Enable the job interruption on first failed test." I had the feeling that this should be only for fails.

Reply:
OK for fail only (I was unsure if that matches the current implementation for the legacy runner).